frontend/src/service-worker.js

```diff
@@ -27,39 +27,62 @@ precacheAndRoute(self.__WB_MANIFEST);
 const fileExtensionRegexp = new RegExp('/[^/?]+\\.[^/]+$');
 registerRoute(
   // Return false to exempt requests from being fulfilled by index.html.
-  ({ request, url }) => {
+  ({request, url}) => {
     // If this isn't a navigation, skip.
     if (request.mode !== 'navigate') {
       return false;
-    } // If this is a URL that starts with /_, skip.
+    }
 
+    // If this is a URL that starts with /_, skip.
     if (url.pathname.startsWith('/_')) {
       return false;
     } // If this looks like a URL for a resource, because it contains // a file extension, skip.
 
     if (url.pathname.match(fileExtensionRegexp)) {
       return false;
-    } // Return true to signal that we want to use the handler.
+    }
 
+    // Return true to signal that we want to use the handler.
     return true;
   },
-  createHandlerBoundToURL(process.env.PUBLIC_URL + '/index.html')
+  createHandlerBoundToURL(process.env.PUBLIC_URL + '/index.html'),
 );
 
-// An example runtime caching route for requests that aren't handled by the
-// precache, in this case same-origin .png requests like those from in public/
-registerRoute(
-  // Add in any other file extensions or routing criteria as needed.
-  ({ url }) => true, // XXX Caching everything for now. It breaks firebase, but it makes most of the PWA work offline. We can narrow it down later.
-  new StaleWhileRevalidate({
-    cacheName: 'everything',
-    plugins: [
-      // Ensure that once this runtime cache reaches a maximum size the
-      // least-recently used stuff is removed.
-      new ExpirationPlugin({ maxEntries: 50 }),
-    ],
-  })
-);
+if (process.env.NODE_ENV === "production" || process.env.NODE_NEV === "dev-pwa") {
+  registerRoute(
+    ({url}) => {
+      const urlString = url.toString();
+      if (urlString.startsWith('https://cdn.jsdelivr.net/')) {
+        return true;
+      } else if (urlString.startsWith('https://cdnjs.cloudflare.com')) {
+        return true;
+      } else if (urlString.startsWith('https://stackpath.bootstrapcdn.com/bootstrap/')) {
+        // TODO: move bootstrap inline rather than getting from CDN
+        return true;
+      } else if (urlString.startsWith('https://futurecoder.io')) {
```

Comment:
Rather just ensure it contains `futurecoder.io`, to account for translations like https://es-latam.futurecoder.io/course/